Break up /database-access/guides/

ptgott opened the pull request:
Closes #17641

Organize the guides to enrolling databases in specific cloud providers (or self-hosted databases) into separate subsections of the "Database Actions" section, organized by cloud provider (or self-hosted).

We can later add general information to the index page of each subsection about how Teleport database access works for each cloud provider. The remaining guides in the "Guides" subsection have to do with database-agnostic functionality of the Teleport Database Service.

This change also adds a redirect for each page it moves.

ptgott committed:
Break up /database-access/guides/

Closes #17641

Organize the guides to enrolling databases in specific cloud providers
(or self-hosted databases) into separate subsections of the "Database
Actions" section, organized by cloud provider (or self-hosted).

We can later add general information to the index page of each
subsection about how Teleport database access works for each cloud
provider. The remaining guides in the "Guides" subsection have to do
with database-agnostic functionality of the Teleport Database Service.

This change also adds a redirect for each page it moves.

greedy52 commented on docs/pages/database-access/enroll-aws-databases.mdx:
```diff
@@ -0,0 +1,30 @@
+- [Amazon RDS Proxy MySQL](./enroll-aws-databases/rds-proxy-mysql.mdx)
+- [Amazon OpenSearch](./enroll-aws-databases/aws-opensearch.mdx)
+- [Amazon DynamoDB](./enroll-aws-databases/aws-dynamodb.mdx)
+- [AWS Keyspaces (Apache Cassandra)](./enroll-aws-databases/aws-cassandra-keyspaces.mdx)
+- [AWS ElastiCache and AWS MemoryDB for Redis](./enroll-aws-databases/redis-aws.mdx)
```
```suggestion
- [Amazon Keyspaces (Apache Cassandra)](./enroll-aws-databases/aws-cassandra-keyspaces.mdx)
- [Amazon ElastiCache and MemoryDB for Redis](./enroll-aws-databases/redis-aws.mdx)
```

greedy52 commented on docs/pages/database-access/introduction.mdx:
```diff
@@ -39,14 +39,38 @@ with GitHub, execute a few SQL queries and observe them in the audit log:
+Teleport can protect both self-hosted and managed databases. Learn how to set up
+secure access to databases in your infrastructure:
+
+- Managed [Amazon Web Services](./enroll-aws-databases.mdx) databases
+- Managed [Google Cloud](./enroll-google-cloud-databases.mdx) databases
+- [Self-Hosted](./enroll-self-hosted-databases.mdx) databases, regardless of
+  cloud provider
+- Managed [Microsoft Azure](./enroll-azure-databases.mdx) databases
+- [Database-specific cloud
+  platforms](docs/pages/database-access//enroll-managed-databases.mdx) such as
+  Snowflake and MongoDB Atlas
```
I left the same comment in other pr but then realized it comes from here. Do we want to link self-hosted here?

ptgott replied:
I've mentioned self-hosted databases on line 54—I'll see if I can separate it out to make it clearer

ptgott committed:
Respond to greedy52 feedback

- In the "Database Access" intro page, move the self-hosted section
  first in the list of links to database-specific guides so it's less
  easy to miss.
- Change "AWS" to "Amazon" in product names.

ptgott committed:
Fix linter issues

ptgott committed:
Remove redirects added by the DB guide reorg

This prevents us from running into Vercel route limit issues, since the
number of redirects is large.
